lib/element-slice.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var _ = require('lodash');
+var uptown = require('uptown');
+var refract = require('./refraction').refract;
+var createClass = uptown.createClass;
+
+var ElementSlice = createClass({
+  // elements: [Element]
+  constructor: function (elements) {
+    this.elements = elements || [];
+  },
+
+  toValue: function () {
+    return this.elements.map(function (element) {
+      return element.toValue();
+    });
+  },
+
+  // High Order Functions
+
+  map: function (callback, thisArg) {
+    return this.elements.map(callback, thisArg);
+  },
+
+  // Filters the element slice
+  // Returns a new sub slice
+  filter: function (callback, thisArg) {
+    return new ElementSlice(this.elements.filter(callback, thisArg));
+  },
+
+  forEach: function (callback, thisArg) {
+    this.elements.forEach(callback, thisArg);
+  },
+
+  reduce: function (callback, initialValue) {
+    return this.elements.reduce(callback, initialValue);
+  },
+
+  includes: function(value) {
+    for (var i = 0; i < this.length; i++) {
+      var item = this.elements[i];
+      if (_.isEqual(item.toValue(), value)) {
+        return true;
+      }
+    }
+
+    return false;
+  },
+
+  // Mutation
+
+  shift: function() {
+    return this.elements.shift();
+  },
+
+  unshift: function(value) {
+    this.elements.unshift(refract(value));
```

Comment:
`refract` is effectively a noop when the element is already Refracted (https://github.com/refractproject/minim/blob/master/lib/refraction.js#L14), it will just return the element.